app/forms/upload_form.rb

```diff
@@ -45,7 +45,14 @@ def save
 
     search_results = keywords.map { |keyword| SearchResult.new(keyword: keyword, search_engine: search_engine) }
     SearchResult.import search_results
+    process_results(search_results)
```

Comment:
You might be interested in using a more modern feature of Ruby to make the code more expressive:

```

def save
  content = csv_file.read
  keywords = CSV.parse(content).flatten.map(&:strip)

  keywords
    .map { |keyword| SearchResult.new(keyword: keyword, search_engine: search_engine) }
    .then(&method(:save_search_results)
	  .then(&method(:scrap_search_results)
end

private

def save_search_results(search_results)
   SearchResult.import(search_results)
end

def scrap_search_results(search_results)
  SearchKeywordJob.perform_later(search_result.id)
end
```

Here is a good article about using `tap` and `then` (available since Ruby 2.6): https://www.ananunesdasilva.com/posts/ruby-methods-tap-vs-then